### ISSUE TITLE
OIDC timeout 3500ms

### ISSUE THREAD

**Issue by depuits** [fmartinou/whats-up-docker#144]
I'm having trouble using authentik to authenticate myself in the app. I'm getting a timed out 3500ms error message. I already found out this is coming from the `openid-client` library and I can fix it by changing the timeout but it would be nice to have this as an environmental variable or something so that it keeps working when updating the image.

In the [app](https://github.com/fmartinou/whats-up-docker/tree/master/app)/[authentications](https://github.com/fmartinou/whats-up-docker/tree/master/app/authentications)/[providers](https://github.com/fmartinou/whats-up-docker/tree/master/app/authentications/providers)/[oidc](https://github.com/fmartinou/whats-up-docker/tree/master/app/authentications/providers/oidc)/Oidc.js I've added following to the top:

```
const { Issuer, generators, custom } = require('openid-client');
const { v4: uuid } = require('uuid');
const Authentication = require('../Authentication');
const OidcStrategy = require('./OidcStrategy');
const { getPublicUrl } = require('../../../configuration');

custom.setHttpOptionsDefaults({
  timeout: 15000,
});
```

This can probably be changed somewhere else and the timeout probably doesn't need to be that high but this is working.

**Comment by fmartinou** [comment]
Hi,

> I already found out this is coming from the openid-client library

Do you have elements for understanding why do you encounter this timeout issue?
Because 3500ms is already a huge default value (I guess that your run WUD & Authentik on the same network) 🤔 .

**Comment by depuits** [comment]
No idea why, when I've first set it up it worked without getting the timeout. I might need to split up the services because now they are running on a single docker host. WUD and Authentik run on the same network and both are accessed through the same reverse proxy. Other services are not having this problem but they use different languages/libraries and might not even have a timeout setup.

**Comment by fmartinou** [comment]
Hi,

It's implemented in version `5.19.0`.
You can now set a `WUD_AUTH_OIDC_{auth_name}_TIMEOUT` to override the default value (5000ms).

[The documentation has been updated](https://github.com/fmartinou/whats-up-docker/issues/144).

**Comment by fmartinou** [comment]
Thanks for the feedback 👍
